src/lib/consensus/proof_of_stake.ml

```diff
@@ -166,19 +166,24 @@ module Make (Inputs : Inputs_intf) :
     module Slot = struct
       include Segment_id
 
+      let ( <= ) x y = UInt32.compare x y <= 0
```

Comment:
Any reason not to just use comparable.make?

Reply:
No reason in particular. Didn't realize `Comparable.Make` would give you infix operations. Isn't that mostly for creating the `Map` submodule?